source/utils/timed-out.ts

```diff
@@ -110,15 +113,15 @@ export default (request: ClientRequest, delays: Delays, options: TimedOutOptions
 
 		/* istanbul ignore next: hard to test */
 		if (socket.connecting) {
-			if (delays.lookup !== undefined && !socketPath && !net.isIP(hostname || host)) {
+			if (delays.lookup !== undefined && !socketPath && !net.isIP(hostname || host || '')) {
```

Comment:
`net.isIP(undefined)` returns `0` but the types for `net.isIP` doesn't allow passing `undefined`.

It doesn't seem impossible that `net.isIP(undefined)` will start throwing `TypeError` in a future version of Node.js, so I think it's better to pass in an empty string which should always return `0`.